modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/modals/selection_filter/SelectionFilter.tsx

```diff
@@ -281,113 +284,122 @@ function Body({
 						)
 				);
-			}
 
-			if (source && sourceType === ESelectionFilterSourceType.PICKLIST) {
-				validSavedPreselectedValues = (
-					source as IPickList
-				).listTypeEntries.filter((item) =>
-					JSON.parse(
-						(filter as ISelectionFilter).preselectedValues || '[]'
-					).includes(item.externalReferenceCode)
+				setPreselectedValues(validSavedPreselectedValues);
+				setIncludeMode(
+					validSavedPreselectedValues?.length
+						? filter && selectionFilter.include
+							? 'include'
+							: 'exclude'
+						: 'include'
 				);
-			}
+			});
+		}
 
-			setPreselectedValues(validSavedPreselectedValues);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
 
-			setIncludeMode(
-				validSavedPreselectedValues?.length
-					? filter && (filter as ISelectionFilter).include
-						? 'include'
-						: 'exclude'
-					: 'include'
-			);
+	useEffect(() => {
+		if (!filter) {
+			return;
 		}
-	}, [filter, filteredSourceItems, source, sourceType]);
 
-	useEffect(() => {
-		if (sourceType === ESelectionFilterSourceType.PICKLIST) {
-			setFilteredSourceItems(
-				!source
-					? []
-					: (source as IPickList).listTypeEntries
-							.filter((item) =>
-								fuzzy.match(preselectedValueInput, item.name)
-							)
-							.map((item) => ({
+		const selectionFilter = filter as ISelectionFilter;
+
+		if (
+			selectionFilter.sourceType === ESelectionFilterSourceType.PICKLIST
+		) {
+			getAllPicklists().then((items) => {
+				if (!items.length) {
+					setNoPicklistsAvailableValidationError(true);
+				}
+				else {
+					setNoPicklistsAvailableValidationError(false);
+					setSourceType(selectionFilter.sourceType);
+
+					const picklist = items.find(
+						(item) =>
+							String(item.externalReferenceCode) ===
+							selectionFilter?.source
+					);
+
+					if (picklist) {
+						setSource(picklist);
+
+						let validSavedPreselectedValues: any[] = [];
+
+						validSavedPreselectedValues =
+							picklist.listTypeEntries.filter((item) =>
+								JSON.parse(
+									selectionFilter.preselectedValues || '[]'
+								).includes(item.externalReferenceCode)
+							);
+
+						const sourceItems = picklist.listTypeEntries.map(
+							(item) => ({
 								label: item.name,
-								value: String(item.externalReferenceCode),
-							}))
-			);
+								value: item.externalReferenceCode,
+							})
+						);
+
+						setFilteredSourceItems(sourceItems);
+						setPreselectedValues(validSavedPreselectedValues);
+						setIncludeMode(
+							validSavedPreselectedValues?.length
+								? filter && selectionFilter.include
+									? 'include'
+									: 'exclude'
+								: 'include'
+						);
+					}
+				}
+			});
 		}
-	}, [preselectedValueInput, source, sourceType]);
 
-	if (
-		sourceType === ESelectionFilterSourceType.API_HEADLESS &&
-		!picklists.length
-	) {
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
+
+	if (noPicklistsAvailableValidationError) {
```

Comment:
This is not a good solution. It just replaces one form of an error with another:

https://github.com/user-attachments/assets/d6ace5d1-f8f6-415b-8713-cdebe3fdfc60